jp/components/grids_templates/virtualization.md

```diff
@@ -124,6 +124,82 @@ public processData() {
 </div>
 }
 
+@@if (igxName === 'IgxTreeGrid') {
+### Remote Filtering Virtualization
+
+To provide remote filtering, you need to subscribe to the [`onFilteringDone`]({environment:angularApiUrl}/classes/@@igTypeDoc.html#onfilteringdone) output so that you make the appropriate request based on the arguments received. Let's use a flat collection as a data source for our Tree Grid by providing a [`primaryKey`]({environment:angularApiUrl}/classes/@@igTypeDoc.html#primarykey) and a [`foreignKey`]({environment:angularApiUrl}/classes/@@igTypeDoc.html#foreignkey).
+
+```html
+<!-- tree-grid-remote-filtering-sample.html -->
+
+<igx-tree-grid #treeGrid [data]="remoteData | async" primaryKey="ID" foreignKey="ParentID" [autoGenerate]="false" width="100%" height="450px"
+                [autoGenerate]="false" (onFilteringDone)="processData()" [allowFiltering]="true">
+    <igx-column [field]="'Name'" dataType="string"></igx-column>
+    <igx-column [field]="'Title'" dataType="string"></igx-column>
+    <igx-column [field]="'Age'" dataType="number"></igx-column>
+    ...
+</igx-tree-grid>
+```
+
+```typescript
+// tree-grid-remote-filtering-sample.ts
+
+public ngAfterViewInit() {
+    this.processData();
+}
+
+public processData() {
+    this.toast.show();
+
+    const filteringExpr = this.treeGrid.filteringExpressionsTree;
+
+    this._remoteService.getData(filteringExpr, () => {
+        this.toast.hide();
+    });
+}
+```
+
+The remote filtering will have to be performed over the flat collection directly. We will also include all the parents for any record that matches the filtering condition regardless of whether or not the parents match the filtering (we do this to keep the hierarchy intact).
+
+```typescript
+// remoteFilteringService.ts
+
+private filterData(data: any[], filteringArgs: IFilteringExpressionsTree): any[] {
```

Comment:
This is a mock-up code which in real life scenario will be performed on the server so it is better to remove this snippet from the docs. We should just mention that when filtering on the server you should include the parents of the matching rows.